lib/supavisor/db_handler.ex

```diff
@@ -307,10 +327,13 @@ defmodule Supavisor.DbHandler do
       # after which it will unlink the direct db connection process from itself.
       data =
         if data.mode == :transaction do
-          ClientHandler.db_status(data.caller, :ready_for_query, bin)
+          # for some reason, if we send the data **before** doing the `ready_for_query` cast,
+          # we get race condition that sometimes causes msgs to be sent to the wrong socket
+          ClientHandler.db_status(data.caller, :ready_for_query)
+          data = intercept_or_send_pkts(pkts, data)
```

Comment:
I don't get why this happens. For me it seems like we want to send stuff and only then call `db_status`. But if I do it it breaks things.